src/intersection-observer.js

```diff
@@ -143,16 +149,19 @@ export class IntersectionObserver {
      * intersection updates for this element.
      * Triggered by context.observeIntersection(…) inside the ad/iframe
      * or by directly posting a send-intersections message.
-     * @private {!SubscriptionApi}
+     * @private {!SimplePostMessageApiDef}
      */
-    this.postMessageApi_ = new SubscriptionApi(
+    this.postMessageApi_ = opt_MessageApi || new SubscriptionApi(
         iframe, 'send-intersections', opt_is3p || false,
         // Each time someone subscribes we make sure that they
         // get an update.
-        () => this.startSendingIntersectionChanges_());
+        () => this.startSendingIntersectionChanges());
```

Comment:
I want to check if we really need to use the `IntersectionObserver` here. I know it's a little messy, but we have `IntersectionObserver` `IntersectionObserverPolyfill` and `IntersectionObserverApi` 😢 `IntersectionObserverPolyfill` is the polyfill to native `IntersectionObserver`. 
Our `IntersectionObserver` and `IntersectionObserverApi` both achieve the same functionality that send message to subscribed iframe on intersection change. The difference here is that `IntersectionObserverApi` limit message count by threshold, while `IntersectionObserver` limit message count by delay. 

If you are using your own `opt_MessageApi`, do you think it's possible that you use the `IntersectionObserverPolyfill` instead? And then set a threshold to only send message when intersection ratio cross the threshold you defined?

Reply:
As discussed offline, I removed my use of IntersectionObserver and switched to using getClientRectAsync.